File: srx/Bench/Extended.fs
module Bench_Extended_FFI

let mutable private extendedResult : obj = null

let consumeResult (expected: obj) (result: obj) =
    box (fun (_: obj) ->
        System.Threading.Volatile.Write(&extendedResult, result)
        if unbox<string> result <> unbox<string> expected then
            failwith "Unstable extended benchmark result"
        box ())

File: srx/Bench/Extended.rs
pub fn Bench_Extended_consumeResult(expected: String, result: String) -> crate::UnknownType {
    crate::Value::Func1(purust_core::Func1::Shared(std::rc::Rc::new(move |_| {
        std::hint::black_box(&result);
        assert_eq!(result, expected, "Unstable extended benchmark result");
        crate::Value::Unit
    })))
}
